qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -186,28 +204,39 @@ def _compute_output_shape(
 
     def _preprocess(
         self,
-        input_data: list[float] | np.ndarray | float,
-        weights: list[float] | np.ndarray | float,
-    ) -> tuple[np.ndarray, int]:
+        input_data: np.ndarray | None,
+        weights: np.ndarray | None,
+    ) -> tuple[np.ndarray | None, int | None]:
         """
         Pre-processing during forward pass of the network.
         """
 
-        np_input_data: np.ndarray = np.asarray(input_data)
-        np_weights: np.ndarray = np.asarray(weights)
-        if len(np_input_data.shape) == 1:
-            np_input_data = np.expand_dims(np_input_data, 0)
-
-        num_samples = max(np_input_data.shape[0], 1)
-        np_weights = np.broadcast_to(np_weights, (num_samples, len(np_weights)))
-        parameters = np.concatenate((np_input_data, np_weights), axis=1)
+        if input_data is not None:
+            num_samples = input_data.shape[0]
+            if weights is not None:
+                weights = np.broadcast_to(weights, (num_samples, len(weights)))
+                parameters = np.concatenate((input_data, weights), axis=1)
+            else:
+                parameters = input_data
+        else:
+            if weights is not None:
+                num_samples = 1
+                parameters = np.broadcast_to(weights, (num_samples, len(weights)))
+            else:
+                return None, None
 
         return parameters, num_samples
 
-    def _postprocess(self, num_samples: int, result: SamplerResult) -> np.ndarray:
+    def _postprocess(
+        self, num_samples: int | None, result: SamplerResult | None
```

Comment:
Won't `num_samples` being None fail below e.g in np.zeros and range.

Reply:
Fixed it, together with comment below.